pages/learn/glossary.md

```diff
@@ -106,6 +112,13 @@ Strictly speaking, according to the specification, schemas are themselves [JSON
 
 In recent [drafts](#draft) of the specification, a schema is either a JSON object or a JSON boolean value.
 
+#### schema resource
+
+A schema resource is a [schema](#schema) which is canonically identified by an absolute URI. This implies all root/parent schemas are schema resources as they can be identified by an absolute URI. 
```

Comment:
> A schema resource is a [schema](#schema) which is canonically identified by an absolute URI.

This is just taken verbatim from [Core 4.3.5](https://json-schema.org/draft/2020-12/json-schema-core#section-4.3.5).  The idea of this glossary is to make things a bit more plain.

> This implies all root/parent schemas are schema resources as they can be identified by an absolute URI.

This isn't precisely true.

```json
{ "type": "object" }
```

is a root schema, but it's not a resource because it's not identifiable.

---

We should say that a schema resource is any schema or subschema which carries an `$id` keyword and then re-iterate the specification's suggestion that all root schemas should carry an `$id` keyword.

Reply:
Greg, I think you've uncovered an inconsistency in the spec. I think that your example has to be considered a Schema Resource. Here's a slightly more complex schema to explain why.

```json
{
  "type": "object",
  "properties": {
    "foo": { "$ref": "https://example.com/schema/foo" }
  },
  "$defs": {
    "foo": {
      "$id": "https://example.com/schema/foo",
      "type": "string"
    }
  }
}
```

The root schema is one of the Schema Resources that makes up this Compound Schema Document, but it doesn't have an identifier. If we don't consider the root schema a Schema Resource, then this wouldn't be a Compound Schema Document, which doesn't make sense. There has to be an exception so that the root schema doesn't need to have `$id`.